src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    public void setup() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
```

Comment:
you initialize the user in every test method, maybe it would be better to do it in the setup method? 
for example with setters.

Reply:
> you initialize the user in every test method, maybe it would be better to do it in the setup method? for example with setters.

Tests must be separated.

Reply:
it is fine to initialize him in the `BeforeEach` method, but not just with field setters, we should also set this user as `new User()` to ensure that it has nothing to do with the previous/next tests
So let's do this

Reply:
Another thing I wanted to mention is do we need a new Service and DAO instances for **each** test?